cli/connhelper/connhelper.go

```diff
@@ -53,3 +56,19 @@ func GetCommandConnectionHelper(cmd string, flags ...string) (*ConnectionHelper,
 		Host: "http://docker",
 	}, nil
 }
+
+func multiplexingArgs() []string {
+	if v := os.Getenv("DOCKER_SSH_NO_MUX"); v != "" {
+		if b, err := strconv.ParseBool(v); err == nil && b {
+			return nil
+		}
+	}
+	if err := os.MkdirAll(config.Dir(), 0700); err != nil {
+		return nil
+	}
+	args := []string{"-o", "ControlMaster=auto", "-o", "ControlPath=" + config.Dir() + "/%r@%h:%p"}
+	if v := os.Getenv("DOCKER_SSH_MUX_PERSIST"); v != "" {
+		args = append(args, "-o", "ControlPersist="+v)
```

Comment:
Is there any reason not to have a Docker config option for this as well?

Reply:
If you would ever set them they would probably be more dependent on the actual `DOCKER_HOST` value, not a global client config. I'm not even sure we even want to document these, it should be very advanced usage if you want to set them.

Reply:
ping @justincormack pal ^